tests/unit/intermediate/test_infer_schema.py

```diff
@@ -0,0 +1,57 @@
+from unittest.mock import create_autospec
+
+import pytest
+from databricks.sdk import WorkspaceClient
+
+from databricks.labs.remorph.intermediate.infer_schema import InferSchema
+
+
+@pytest.fixture(autouse=True)
+def setup_file(tmpdir):
+    file = tmpdir.join("test.sql")
+    file.write(
+        """SELECT a.col1, sum(b.col2) as col2, to_date(col11) 
```

Comment:
Design is flawed. Our code has to expect a collection of strings and do the work with them. It's the implementation detail where those strings are coming from - files or query history or something else.

Rewrite the tests and implementation